src/maker.rs

```diff
@@ -207,13 +200,17 @@ pub enum TakeRequestDecision {
 
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub enum RateUpdateDecision {
-    RateUpdatedWithNewValue {
-        next_sell_order: BtcDaiOrder,
-        next_buy_order: BtcDaiOrder,
+    RateChange {
+        new_sell_order: BtcDaiOrder,
+        new_buy_order: BtcDaiOrder,
     },
-    RateUpdated,
+    NoRateChange,
 }
 
+#[derive(Debug, Copy, Clone, thiserror::Error)]
+#[error("Rate not available when trying to create new {0} order.")]
+pub struct RateNotAvailable(Position);
```

Comment:
:wrench: I don't think the position adds any useful context.

Reply:
I thought it might add value if this ever happens in `process_taken_order` - don't have strong feelings about it though.